zqd/space/space.go

```diff
@@ -34,59 +36,207 @@ var (
 	ErrSpaceNotExist       = zqe.E(zqe.NotFound, "space does not exist")
 )
 
+type Manager struct {
+	rootPath string
+	spaces   sync.Map
```

Comment:
sync.Map is not the right tool for this job. As https://pkg.go.dev/sync?tab=doc#Map says:

> The Map type is specialized. Most code should use a plain Go map instead, with separate locking or coordination, for better type safety and to make it easier to maintain other invariants along with the map content.

That's borne out by the race in Create.